caddyhttp/push/handler.go

```diff
@@ -25,7 +26,13 @@ func (h Middleware) ServeHTTP(w http.ResponseWriter, r *http.Request) (int, erro
 	// push first
 outer:
 	for _, rule := range h.Rules {
-		if httpserver.Path(r.URL.Path).Matches(rule.Path) {
+		urlPath := r.URL.Path
+		matches := httpserver.Path(urlPath).Matches(rule.Path)
+		// Also check IndexPages when requesting a directory
+		if !matches {
+			_, matches = httpserver.IndexFile(h.Root, urlPath, staticfiles.IndexPages)
```

Comment:
Ahh, yeah, duh, how did we miss that. We should add a test case to check for this. :)